src/sys/socket/mod.rs

```diff
@@ -627,7 +633,9 @@ pub unsafe fn sockaddr_storage_to_addr(
             Ok(SockAddr::Inet(InetAddr::V6((*(addr as *const _ as *const sockaddr_in6)))))
         }
         consts::AF_UNIX => {
-            Ok(SockAddr::Unix(UnixAddr(*(addr as *const _ as *const sockaddr_un), len)))
```

Comment:
The old code forgot to consider the offset of sun_path.  The result was that getsockname would appear to return a path with one or two extra null bytes at the end.